### PR TITLE
WIP: Switch region

### PR DESCRIPTION
**Pull request by outscale-vbr**
**What type of PR is this?**
/kind cleanup    

**What this PR does / why we need it**:
switch region to us-east-2
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

### CODE REVIEW

File: .github/workflows/image-builder.yaml [hunk 1]
```diff
@@ -37,9 +37,9 @@ jobs:
       shell: bash
       working-directory: "${{ github.workspace }}/add-outscale-image/images/capi"
       env:
-         OSC_ACCESS_KEY: ${{secrets.OSC_ACCESS_KEY}}
-         OSC_SECRET_KEY: ${{secrets.OSC_SECRET_KEY}}   
-         OUTSCALE_REGION: ${{secrets.OSC_REGION}}
+         OSC_ACCESS_KEY: ${{secrets.OSC_OPENSOURCE_ACCESS_KEY}}
+         OSC_SECRET_KEY: ${{secrets.OSC_OPENSOURCE_SECRET_KEY}}   
+         OUTSCALE_REGION: ${{secrets.OSC_OPENSOURCE_REGION}}
```
Comment (outscale-hmi) [inline .github/workflows/image-builder.yaml:42]: why OUTSCALE_REGION or OSC_REGION

**Review by outscale-hmi — COMMENTED** [review]
Good, just one comment about region env var ?

File: .github/workflows/image-builder.yaml (hunk 1 above)
Comment (outscale-vbr) [reply .github/workflows/image-builder.yaml:42]: https://www.packer.io/plugins/builders/outscale
In packer docs it is OUTSCALE_REGION .
we have to change to  OSC_REGION ?

Comment (outscale-hmi) [reply .github/workflows/image-builder.yaml:42]: yes use OSC_REGION , the release of updated doc is not published yet

File: .github/workflows/unit-func-e2e-test.yaml [hunk 2]
```diff
@@ -192,6 +174,21 @@ jobs:
...
+  unit_test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          ref: ${{ github.event.pull_request.head.ref }}
+    - uses: actions/setup-go@v3
+      with:
+        go-version: '1.18.5'
```
Comment (outscale-hmi) [inline .github/workflows/unit-func-e2e-test.yaml:186]: think about go-version-file later ?

**Review by outscale-hmi — COMMENTED** [review]
ok

File: .github/workflows/unit-func-e2e-test.yaml (hunk 2 above)
Comment (outscale-vbr) [reply .github/workflows/unit-func-e2e-test.yaml:186]: I will do in other ticket.

**Comment by outscale-vbr** [comment]
https://github.com/outscale-dev/cluster-api-provider-outscale/pull/103 must be merged first.

**Comment by outscale-vbr** [comment]
close https://github.com/outscale-dev/cluster-api-provider-outscale/issues/101

File: api/v1beta1/types.go [hunk 3]
```diff
@@ -297,10 +297,11 @@ type OscVolume struct {
...
 type OscKeypair struct {
-	Name        string `json:"name,omitempty"`
-	PublicKey   string `json:"publicKey,omitempty"`
-	ResourceId  string `json:"resourceId,omitempty"`
-	ClusterName string `json:"clusterName,omitempty"`
+	Name          string `json:"name,omitempty"`
+	PublicKey     string `json:"publicKey,omitempty"`
+	ResourceId    string `json:"resourceId,omitempty"`
+	ClusterName   string `json:"clusterName,omitempty"`
+	DeleteKeypair bool   `json:"deleteKeypair,omitempty"`
```
Comment (outscale-mdr) [inline api/v1beta1/types.go:304]: For another PR ?

File: cloud/scope/machine.go [hunk 4]
```diff
@@ -171,6 +171,16 @@ func (m *MachineScope) GetKeypair() *infrastructurev1beta1.OscKeypair {
...
+// SetDeleteKeyPair set deleteKeypair
+func (m *MachineScope) SetDeleteKeypair(deleteKeypair bool) {
+	m.OscMachine.Spec.Node.KeyPair.DeleteKeypair = deleteKeypair
+}
+
+// GetDeleteKeyPair return deleteKeypair
+func (m *MachineScope) GetDeleteKeypair() bool {
+	return m.OscMachine.Spec.Node.KeyPair.DeleteKeypair
+}
+
```
Comment (outscale-mdr) [inline cloud/scope/machine.go:183]: Same ?

**Review by outscale-mdr — CHANGES_REQUESTED** [review]
Please, could you rework this PR. Only one commit with different features.

**Review by outscale-hmi — APPROVED** [review]
OK

**Comment by outscale-vbr** [comment]
Closes https://github.com/outscale-dev/cluster-api-provider-outscale/issues/101

**Pull request merged**